resolwe/storage/manager.py

```diff
@@ -18,251 +18,307 @@
 class DecisionMaker:
     """Decide which data should be copied to another location or deleted."""
 
-    def __init__(self, file_storage: FileStorage):
-        """Initialization."""
-        self.file_storage = file_storage
-        self.process_type = ""
-        self.data_slug = ""
-        data = file_storage.data.first()
-        if data is not None:
-            self.process_type = data.process.type
-            self.data_slug = data.slug
-
-    def _get_data_slug_overrides(self, override_rules: dict) -> dict:
-        """Get a matching override rule for data slug."""
-        return override_rules.get(self.data_slug, dict())
-
-    def _get_process_type_overrides(self, override_rules: dict) -> dict:
-        """Get a matching override rule for process type."""
-        rule = dict()
-        matching_keys = []
-        for key in override_rules.keys():
-            modified_key = key if key.endswith(":") else key + ":"
-            if self.process_type.startswith(modified_key):
-                matching_keys.append(key)
-        matching_keys.sort(key=len)
-        for matching_key in matching_keys:
-            rule.update(override_rules[matching_key])
-        return rule
-
-    def _get_rules(self, connector_name: str, rule_type: str) -> dict:
-        """Get rules for the given connector name and _type.
-
-        Known rule types are 'copy' and 'delete'.
-        """
-        location_settings = copy.deepcopy(STORAGE_CONNECTORS.get(connector_name, {}))
-        rules: dict = location_settings.get("config", {}).get(rule_type, {})
-        override_rules = []
-        override_rules.append(
-            self._get_process_type_overrides(rules.pop("process_type", {}))
-        )
-        override_rules.append(self._get_data_slug_overrides(rules.pop("data_slug", {})))
-        for override_rule in override_rules:
-            rules.update(override_rule)
-        return rules
-
-    def copy(self) -> List[str]:
-        """Get a list of connector names where data must be copied to."""
-        return [
-            connector_name
-            for connector_name in STORAGE_CONNECTORS
-            if self._should_copy_to(connector_name)
-        ]
-
-    def _should_copy_to(self, connector_name: str) -> bool:
-        """Get if data should be copied to the given location.
+    def _preprocess_delay(self, delay: Optional[int]) -> int:
+        """Preprocess delay according to rules.
 
-        :param connector_name: name of the storage connector.
-        :type connector_name: str
-
-        :return: True when data should be copied to the location managed by
-            connector connector_name.
+        1. When delay is None return 0. Entries without explicit delay are
+            copied immediately.
```

Comment:
This method is also used for deleting locations, so it should not use "copy". Use "process" instead.

Reply:
Done.